selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/preferences/ContributeTabTest.java

```diff
@@ -0,0 +1,288 @@
+    pullRequestPanel.waitTetxNotVcsProject(EXP_TEXT_NOT_VCS);
+
+    // check the 'Contribute' checkbox is true by default
+    openContributeTab();
+    preferences.waitContributeCheckboxIsSelected();
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkRefreshAndSaveButton() {
+    openContributeTab();
+
+    preferences.setAndWaitStateContributeCheckbox(true);
+    preferences.clickOnOkBtn();
+
+    assertFalse(preferences.saveButtonIsEnabled());
+
+    // check the 'Refresh button
+    preferences.clickOnContributeCheckbox();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    assertTrue(preferences.saveButtonIsEnabled());
+
+    preferences.clickRefreshButton();
+    preferences.waitContributeCheckboxIsSelected();
+
+    assertFalse(
+        preferences.saveButtonIsEnabled(),
+        "Known issue https://github.com/eclipse/che/issues/9959");
+
+    preferences.clickOnCloseBtnAndWaitClosedForm();
+  }
+
+  @Test(priority = 1)
+  public void checkSwitchProjectsWhenContributeIsFalse() {
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(true);
+
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(false);
+
+    // switch between projects
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitRepoUrl(firstProjectUrl);
+    pullRequestPanel.waitProjectName(FIRST_PROJECT_NAME);
+
+    projectExplorer.waitAndSelectItem(SECOND_PROJECT_NAME);
+    pullRequestPanel.waitRepoUrl(secondProjectUrl);
+    pullRequestPanel.waitProjectName(SECOND_PROJECT_NAME);
+
+    projectExplorer.waitAndSelectItem(THIRD_PROJECT_NAME);
+    pullRequestPanel.waitTetxNotVcsProject(EXP_TEXT_NOT_VCS);
+  }
+
+  @Test(priority = 1)
+  public void checkAutomaticChangeContributeToFalse() {
+    openContributeTab();
+    setStateContributeChecboxAndCloseForm(true);
+
+    // change 'Contribute' to false by 'Hide' button
+    projectExplorer.waitAndSelectItem(FIRST_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+    pullRequestPanel.clickHideButtonAndWaitClosePanel();
+    notificationsPopupPanel.waitExpectedMessageOnProgressPanelAndClosed(NOTIFICATION_MESS);
+
+    openContributeTab();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    // change 'Contribute' to false by 'Hide' from 'Options' on the PR panel
+    setStateContributeChecboxAndCloseForm(true);
+
+    projectExplorer.waitAndSelectItem(SECOND_PROJECT_NAME);
+    pullRequestPanel.waitOpenPanel();
+    pullRequestPanel.openOptionsMenu();
+    pullRequestPanel.clickContextHideButtonAndWAitClosedPanel();
+    notificationsPopupPanel.waitExpectedMessageOnProgressPanelAndClosed(NOTIFICATION_MESS);
+
+    openContributeTab();
+    preferences.waitContributeCheckboxIsNotSelected();
+
+    // change 'Contribute' to false by 'Contribute' action from 'Assistant'
```

Comment:
`change 'Contribute' to false by 'Contribute' action from 'Assistant'` is hard to understand